home/models.py

```diff
@@ -1388,14 +1404,41 @@ def save_revision(
         )
 
         try:
-            template_name = self.submit_whatsapp_template(previous_revision)
+            if not settings.WHATSAPP_CREATE_TEMPLATES:
+                return
+
+            # If there are any missing fields in the previous revision, then carry on
+            if previous_revision:
+                previous_revision = previous_revision.as_object()
+                previous_revision_fields = previous_revision.fields
+            else:
+                previous_revision_fields = ()
+
+            if self.fields == previous_revision_fields:
+                return
+
+            self.template_name = self.create_whatsapp_template_name()
+
+            result_str = create_standalone_whatsapp_template(
+                name=self.template_name,
+                message=self.message,
+                category=self.category,
+                locale=self.locale,
+                quick_replies=sorted(self.quick_reply_buttons),
+                image_obj=self.image,
+                example_values=[v["value"] for v in self.example_values.raw_data],
+            )
+
         except Exception:
             # Log the error to sentry and send error message to the user
             logger.exception(f"Failed to submit template name:  {self.name}")
             raise ValidationError("Failed to submit template")
 
-        if template_name:
-            revision.content["name"] = template_name
+        if result_str:
+            revision.content["name"] = self.name
+            revision.content["submission_name"] = self.template_name
+            revision.content["submission_status"] = self.SubmissionStatus.SUBMITTED
```

Comment:
Does this mean that whether there's an error or not, we're always setting the status to submitted? Should we not be using the failed status for errors?

Reply:
Sorry, I meant to come back to that, but somehow forgot